<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.11"/>
<title>Woolz Image Processing: Libraries</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript" src="navtreedata.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<script type="text/javascript">
  $(document).ready(initResizable);
  $(window).load(resizeHeight);
</script>
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    extensions: ["tex2jax.js"],
    jax: ["input/TeX","output/HTML-CSS"],
});
</script><script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">Woolz Image Processing
   &#160;<span id="projectnumber">Version 1.7.5</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.11 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li class="current"><a href="pages.html"><span>Related&#160;Pages</span></a></li>
      <li><a href="modules.html"><span>Modules</span></a></li>
      <li><a href="annotated.html"><span>Data&#160;Structures</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
$(document).ready(function(){initNavTree('libraries.html','');});
</script>
<div id="doc-content">
<div class="header">
  <div class="headertitle">
<div class="title">Libraries </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><h1><a class="anchor" id="libraries_intro"></a>
Introduction</h1>
<p>The core woolz image processing library, <a class="el" href="libwlzpage.html">libWlz</a>, was developed for the purpose of microscope slide scanning. libWlz holds the basic image processing functionality, supplementary libraries provide functionality for memory allocation and management, numerical algorithms, external format conversion and other commonly used programmatic tools. These are: </p><ul>
<li><a class="el" href="libalcpage.html">libAlc</a> - memory allocation and management </li>
<li><a class="el" href="libalgpage.html">libAlg</a> - numerical algorithms </li>
<li><a class="el" href="libwlzextpage.html">libWlzExt</a> - image format conversion and meta-data IO </li>
<li><a class="el" href="libbibfilepage.html">libbibfile</a> - IO if "bibtex" format data - standard plain-text meta-data </li>
<li><a class="el" href="libdlplistpage.html">libDlpList</a>- linked-list structures and procedures for management, sorting and query.</li>
</ul>
<p>The original woolz software set included libraries for classification and chromosome analysis, these are not included in this package. All libraries are developed in ANSI Standard C and have been successfully used on any Unix system tried so far including Solaris, Linux, IRIX, HPUX, MacOSX. It is also availbale under Windows either using Cygwin or directly using the Microsoft C++ compiler. In addition there is a Java mapping available for the woolz library to allow image process from Java code.</p>
<h1><a class="anchor" id="Structure"></a>
Overview</h1>
<p>Woolz has been generated over many years at the Medical Research Council Human Genetics Unit in Edinburgh. The basic image processing library originally developed has been extended to include a range of numerical algorithms and libraries for standard programming tasks. The code has been converted to ANSI standard C and standardised with respect to nameing conventions and error reporting. Finally the code has been checked for memory errors such array bound exceptions and leaked bytes. The code is maintained by a number of developers at the MRC and version control is under CVS. The code is open-source and freely available for academic non-profit use. For commercial use of the software please contact <a href="#" onclick="location.href='mai'+'lto:'+'mat'+'ec'+'h@h'+'gu'+'.mr'+'c.'+'ac.'+'uk'; return false;">matec<span style="display: none;">.nosp@m.</span>h@hg<span style="display: none;">.nosp@m.</span>u.mrc<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>uk</a>.</p>
<p>The libraries are organised to hierarchically and need to be linked accordingly. The dependencies are: </p><ul>
<li>libAlc - none, system library libc </li>
<li>libAlg - libAlc, system library libm </li>
<li>libbibfile - libAlc </li>
<li>libhguDlpList - libAlc </li>
<li>libWlz - libAlg </li>
<li>libWlzExtFF - libWlz, libbibfile, external libraries libtiff and libjpeg</li>
</ul>
<h1><a class="anchor" id="install"></a>
Installation</h1>
<p>The source code and/or compiled versions of the libraries can be downloaded from the Mouse Atlas Web site (<a href="http://www.emouseatlas.org/emap/analysis_tools_resources/software.html">http://www.emouseatlas.org/emap/analysis_tools_resources/software.html</a>). The installation instructions are provided as part of a "README.txt" file which requires the selection of a path (e.g. /opt/MouseAtlas) as the root of all Mouse Atlas software. The code is then installed in standard Unix fashion in src, share, lib, bin and so on.</p>
<h1><a class="anchor" id="history"></a>
Software Development History</h1>
<p>The Woolz software has been developed over many years by many people with its roots in chromosome spread image analysis and microscopy imaging in the 1970s. The initial development by Dr Denis Rutovitz was in Fortran and defined a number of the key woolz structures, primarily the interval coding of the spatial domains and grey-level values. The current C-coding was initiated in the 1980s, primarily by Dr Jim Piper. The coding has been transformed to its present form in a number of steps as part of early Alvey and EU funded projects and more recently as part of the Mouse Atlas porject at the HGU. This has include a large expansion in some of the grey-level image extgensions to the library. The Woolz image processing library is optimised for efficient for binary and morphological operations. The software has been developed in ANSI standard C.</p>
<p>Originally the code was developed by the Pattern Recognition Section of the Medical Research Council (<a href="http://www.mrc.ac.uk/">MRC</a>) Clinical and Population Cytogenetics Unit, since 1995 it has been developed as part of the Edinburgh Mouse Atlas Project (<a href="http://www.emouseatlas.org/">EMAP</a>) at the MRC Human Genetics Unit (<a href="http://www.hgu.mrc.ac.uk/">HGU</a>). The Woolz software is a series of libraries supporting a range of image processing operations and is built into a number of larger applications such as MAPaint, Reconstruct and the Jave JAtlasViewer. It also underpins the Edinburgh Mouse Atlas of Gene Expression (EMAGE) database defining the spatial content and providing the mapping, comparison, analysis and warping capabilities.</p>
<p>The current Woolz software is primarily maintained by Richard Baldock and Bill Hill at the MRC Human Genetics Unit.</p>
<h1><a class="anchor" id="pubs"></a>
Publications</h1>
</div></div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="navelem"><a class="el" href="index.html">Woolz Core Software</a></li>
    <li class="footer">Generated by
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.8.11 </li>
  </ul>
</div>
</body>
</html>
